src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,85 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useCallback, useMemo, useState } from 'react';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
 import './FindMovie.scss';
 
-export const FindMovie: React.FC = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+  movies: Movie[];
+};
+
+export const FindMovie: React.FC<Props> = React.memo(({ onAdd, movies }) => {
+  const [query, setQuery] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [foundMovie, setFoundMovie] = useState<Movie | null>(null);
+  const [hasError, setHasError] = useState(false);
+
+  const defaultImgUrl = useMemo(() => {
+    return 'https://via.placeholder.com/360x270.png?text=no%20preview';
+  }, []);
+
+  const clearForm = useCallback(() => {
+    setQuery('');
+    setFoundMovie(null);
+    setHasError(false);
+  }, []);
+
+  const handleSubmit = useCallback(
+    async (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      setIsLoading(true);
+      setHasError(false);
+
+      const newMovie = await getMovie(query);
+
+      try {
+        if ('Error' in newMovie) {
+          setHasError(true);
+        } else {
+          const {
+            Title,
+            Plot,
+            Poster,
+            imdbID,
+          } = newMovie;
+
+          setFoundMovie({
+            title: Title,
+            description: Plot,
+            imgUrl: Poster === 'N/A'
+              ? defaultImgUrl
+              : Poster,
+            imdbUrl: `https://www.imdb.com/title/${imdbID}`,
+            imdbId: imdbID,
+          });
+        }
+      } catch {
```

Comment:
```suggestion
      } catch (error) {
```

Reply:
It's not neccesary to get this argument, if i'm not going to use it later.